gapic-generator/lib/gapic/model/method/http_annotation.rb

```diff
@@ -35,122 +43,174 @@ class HttpAnnotation
-        # @return [Array<String>]
-        def routing_params
-          routing_params_with_patterns.map { |param, _| param }
-        end
+          ##
+          # The segment key names.
+          #
+          # @return [Array<String>]
+          def routing_params
+            routing_params_with_patterns.map { |param, _| param }
+          end
 
-        ##
-        # Whether method has body specified in proto
-        #
-        # @return [Boolean]
-        def body?
-          !body.empty?
-        end
+          ##
+          # Whether method has body specified in proto
+          #
+          # @return [Boolean]
+          def body?
+            !body.empty?
+          end
 
-        ##
-        # The body specified for the given method in proto
-        # or an empty string if not specified
-        #
-        # @return [String]
-        def body
-          http&.body || ""
+          ##
+          # The body specified for the given method in proto
+          # or an empty string if not specified
+          #
+          # @return [String]
+          def body
+            @binding&.body || ""
+          end
+
+          private
+
+          ##
+          # The combination of verb and path found in the http annotation
+          # (or Nil if the annotation is Nil).
+          #
+          # @return [Array<Symbol, String>, Nil]
+          def verb_path
+            return nil if @binding.nil?
+
+            {
+              get:    @binding.get,
+              post:   @binding.post,
+              put:    @binding.put,
+              patch:  @binding.patch,
+              delete: @binding.delete
+            }.find { |_, value| !value.empty? }
+          end
         end
 
         private
 
         ##
-        # The Http annotation to use -- an override from the service config
-        # or the proto method's one
+        # Parses http bindings from the `google.api.http` annotation, preserving order.
         #
-        # @return [::Google::Api::Http, Nil]
-        def http
-          @http_override || @proto_http
-        end
-
-        ##
-        # The combination of verb and path found in the http annotation
-        # (or Nil if the annotation is Nil).
+        # @param http [::Google::Api::HttpRule]
+        #   The root instance of the HttpRule to parse.
         #
-        # @return [Array<Symbol, String>, Nil]
-        def verb_path
-          return nil if http.nil?
-
-          method = {
-            get:    http.get,
-            post:   http.post,
-            put:    http.put,
-            patch:  http.patch,
-            delete: http.delete
-          }.find { |_, value| !value.empty? }
-
-          method unless method.nil?
+        # @return Array<HttpBinding>
+        #  The parsed bindings.
+        def parse_bindings http
+          binds = []
+          return binds if http.nil?
+
+          raw_binds = [http]
+          if http.additional_bindings&.any?
+            raw_binds += http.additional_bindings.to_a
```

Comment:
`nil.to_a == []` so I think you can dispense with the if statement. Unless there are nil array members?

Reply:
:+1: